snuba/subscriptions/executor_consumer.py

```diff
@@ -446,7 +446,7 @@ def poll(self) -> None:
             self.__queue.popleft()
 
             self.__commit_data[message.partition] = Position(
-                message.offset, message.timestamp
+                message.next_offset, message.timestamp
```

Comment:
Some questions:

1. What is the impact of this being incorrect right now?

2. This `next_offset` behavior appears in three separate places. Is there a way to make it so that it can't be misused? This seems like special behavior and its knowledge is duplicated. At the very least, I'd like there to be a comment in each one explaining at each place why this is happening because this is not intuitive

3. How and where is `next_offset` calculated? is it just `message.offset + 1`?

Reply:
1. We are always committing one offset behind what we should be. In practice it means when we rebalance / restart consumers we are likely to be double processing the last processed message. Subscriptions is an at least once system so it is considered harmless that we run it twice.
2. It appears in 3 different places because I was unaware that the correct offset to be committed is the next one and was committing the current one everywhere.
3.  Yes it is `message.offset + 1`. Somewhat related: https://github.com/getsentry/rust_arroyo/pull/18 and https://github.com/getsentry/arroyo/pull/73

Reply:
This is documented in https://github.com/getsentry/arroyo/blob/main/arroyo/backends/abstract.py#L14-L49

Reply:
Ah I see, so `next_offset` what is always meant to be committed to kafka, not just for subscriptions. That was where I was confused.